Set d0i3 register on MTL

RanderWang opened the pull request:
All cavs platform use the same d0i3 register offset but MTL doesn't. This PR addresses this issue by introducing d0i3_offset in chip_info

ujfalusi commented on sound/soc/sof/ops.h:
```diff
@@ -324,6 +324,27 @@ static inline void snd_sof_dsp_write64(struct snd_sof_dev *sdev, u32 bar,
+		return sof_ops(sdev)->readb(sdev, sdev->bar[bar] + offset);
+
+	dev_err(sdev->dev, "error: %s not defined\n", __func__);
+	return (char)-EOPNOTSUPP;
```
The caller will have hard time to figure out that the u8 casted error code is a value or error, Just return 0?
Yes, we shall fix this for the other variants as well.

ujfalusi commented on sound/soc/sof/intel/mtl.h:
```diff
@@ -21,6 +21,8 @@
 #define MTL_IRQ_INTEN_L_SOUNDWIRE_MASK	BIT(6)
 #define MTL_HFINTIPPTR_PTR_MASK		GENMASK(20, 0)
 
+#define MTL_HDA_VS_D0I3C	0x1D4A
```
One more tab to align it?

ujfalusi commented on sound/soc/sof/intel/hda-dsp.c:
```diff
@@ -378,27 +381,31 @@ static int hda_dsp_send_pm_gate_ipc(struct snd_sof_dev *sdev, u32 flags)
 
 	/* Update D0I3C register */
-	snd_hdac_chip_updateb(bus, VS_D0I3C, SOF_HDA_VS_D0I3C_I3, value);
+	snd_sof_dsp_writeb(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset, value);
```
updateb is not equivalent to writeb!

RanderWang replied:
o, thanks

plbossart commented:
@RanderWang it seems we both worked on the same topic, I added readb/writeb capabilities and modified the D0i3C yesterday :-) see https://github.com/thesofproject/linux/pull/3884

RanderWang commented:
> @RanderWang it seems we both worked on the same topic, I added readb/writeb capabilities and modified the D0i3C yesterday :-) see #3884

thanks! I will close it

plbossart commented:
I'll extract to patches to a new PR so that you are not blocked @RanderWang

RanderWang commented:
> I'll extract to patches to a new PR so that you are not blocked @RanderWang

Thanks! I can enjoy my vocation : )
